x/pki/types/message_add_pki_revocation_distribution_point_test.go

```diff
@@ -0,0 +1,488 @@
+				Label:  "label",
+			},
+			err: validator.ErrRequiredFieldMissing,
+		},
+		{
+			name: "issuerSubjectKeyId empty",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+			},
+			err: validator.ErrRequiredFieldMissing,
+		},
+		{
+			name: "dataUrl empty",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+			},
+			err: validator.ErrRequiredFieldMissing,
+		},
+		{
+			name: "revocationType empty",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+			},
+			err: validator.ErrRequiredFieldMissing,
+		},
+		{
+			name: fmt.Sprintf("dataDigestType is not one of %v", allowedDataDigestTypes),
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+				DataDigestType:       3,
+			},
+			err: pkitypes.ErrInvalidDataDigestType,
+		},
+		{
+			name: fmt.Sprintf("revocationType is not one of %v", allowedRevocationType),
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       2,
+			},
+			err: pkitypes.ErrInvalidRevocationType,
+		},
+		{
+			name: "pid not empty when IsPAA is true",
+			msg: MsgAddPkiRevocationDistributionPoint{
+				Signer:               sample.AccAddress(),
+				Vid:                  1,
+				IsPAA:                &true_,
+				CrlSignerCertificate: testconstants.RootCertPem,
+				Label:                "label",
+				DataUrl:              testconstants.DataURL,
+				IssuerSubjectKeyID:   testconstants.SubjectKeyIDWithoutColons,
+				RevocationType:       1,
+				Pid:                  1,
+			},
+			err: pkitypes.ErrNotEmptyPid,
+		},
+		{
+			name: "dataUrl starts not with http or https",
```

Comment:
Please also add a test (and I suspect the implementation) for the case, when URL is valid, but starts with something  different, for example `ftp://...`.